aspnetcore/fundamentals/websockets.md

```diff
@@ -116,7 +116,17 @@ If using the WebSocket support in [socket.io](https://socket.io/) on [Node.js](h
 
 ## Next steps
 
-The [sample app](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/websockets/sample) that accompanies this article is an echo app. It has a web page that makes WebSocket connections, and the server resends any messages it receives back to the client. Run the app from a command prompt (it's not set up to run from Visual Studio with IIS Express) and navigate to http://localhost:5000. The web page shows the connection status in the upper left:
+::: moniker range=">= aspnetcore-2.0"
```

Comment:
You don't need to version this ... just point the link to the samples folder and let the reader click from there. They'll see that there are 1.x and 2.x samples.

> \[sample app](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/websockets/samples)

What you do need to version is any place in the topic where the sample code is pulled in with links to snippets.

Reply:
Like this one ...

```
[!code-csharp[](websockets/sample/Startup.cs?name=Echo)]
```

That should be versioned for 1.x and 2.x with the moniker ranges.

```
::: moniker range=">= aspnetcore-2.0"

[!code-csharp[](websockets/samples/2.x/WebSocketSample/Startup.cs?name=Echo)]

::: moniker-end

::: moniker range="< aspnetcore-2.0"

[!code-csharp[](websockets/samples/1.x/WebSocketSample/Startup.cs?name=Echo)]

::: moniker-end
```

Reply:
Even when the code doesn't change. The reason is to harden these code snippets for future changes. Since we go this way, we can come back into this 2.x sample (at 2.1, 2.2, etc.) and change whatever we need to without breaking the 1.x version.

Reply:
thanks